pkg/internal/checkup/vmi/vmi.go

```diff
@@ -0,0 +1,289 @@
+/*
+ * This file is part of the kiagnose project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package vmi
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	kvcorev1 "kubevirt.io/api/core/v1"
+)
+
+// Based on annotation names from:
+// https://github.com/cri-o/cri-o/blob/fa0fa5de1c17ddd7b6fcdbc030b6b571ce37e643/pkg/annotations/annotations.go
+const (
+	// CRIOCPULoadBalancingAnnotation indicates that load balancing should be disabled for CPUs used by the container
+	CRIOCPULoadBalancingAnnotation = "cpu-load-balancing.crio.io"
+
+	// CRIOCPUQuotaAnnotation indicates that CPU quota should be disabled for CPUs used by the container
+	CRIOCPUQuotaAnnotation = "cpu-quota.crio.io"
+
+	// CRIOIRQLoadBalancingAnnotation indicates that IRQ load balancing should be disabled for CPUs used by the container
+	CRIOIRQLoadBalancingAnnotation = "irq-load-balancing.crio.io"
+)
+
+const Disable = "disable"
+
+type Builder struct {
```

Comment:
I do not understand the need of this builder object.
The builder pattern was used in other places as a simple factory (without objects), why this is different here? Or more accurately, what does this object comes to solve exactly (compared to the flat function option)?